### PR TITLE
Printlns in simulation.rs are changed to debugs.

### LINKED ISSUE #5: Simulation prints a lot of stuff into stdout
In `simulation.rs`, methods `proceed_if_needed` and `handle_incoming` print a lot, especially if the number of parties is large. This is done via `println!` and thus not very configurable, complicating debug, testing and benchmarking. Maybe we can use `info!` or `debug!` instead? I can make a PR for this

### PR DESCRIPTION
**Pull request by DmytroTym**

### CODE REVIEW

**Comment by survived** [comment]
Does it compile on your machine? I have this error:

```
error: unexpected end of macro invocation
   --> src/dev/simulation.rs:197:13
    |
197 |             debug!();
    |             ^^^^^^^^^ missing tokens in macro arguments
```

**Comment by DmytroTym** [comment]
Oops, sorry!

**Comment by survived** [comment]
Great! Do you think you could also enable logging in [integration tests]? Just to demonstrate that it's possible to trace state changes through the execution.

[integration tests]: https://github.com/ZenGo-X/round-based-protocol/blob/main/tests/simulate_silly_protocol.rs

**Comment by DmytroTym** [comment]
Only enabled it for the first test since the other two do not use simulation.

File: Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,8 @@ futures = { version = "0.3.9", optional = true }
 async-stream = { version = "0.3.0", optional = true }
 thiserror = "1.0.23"
 serde = { version = "1.0", features = ["derive"] }
+log = "0.4.14"
+env_logger = "0.9.0"
```
Comment (survived) [inline Cargo.toml]: We need `env_logger` only for tests, maybe move to `[dev-dependencies]`?

Comment (DmytroTym) [reply Cargo.toml]: Right!

**Comment by survived** [comment]
Great, thanks!

**Pull request merged**